scripts/data_prep/convert_delta_to_json.py

```diff
@@ -0,0 +1,451 @@
+            )
+
+    os.makedirs(args.json_output_path, exist_ok=True)
+
+    log.info(f'Directory {args.json_output_path} created.')
+
+    method = ''
+    dbsql = None
+    sparkSession = None
+
+    if hasattr(args, 'http_path') and args.http_path:
+        method = 'dbsql'
+        try:
+            dbsql = sql.connect(
+                server_hostname=args.DATABRICKS_HOST,
+                http_path=args.http_path,
+                access_token=args.DATABRICKS_TOKEN,
+            )
+        except Exception as e:
+            raise RuntimeError(
+                'Failed to create sql connection to db workspace. Check {server_hostname} and {http_path} and access token!'
+            ) from e
+    else:
+        method = 'dbconnect'
+        if not args.cluster_id:
+            session_id = str(uuid4())
+            sparkSession = DatabricksSession.builder.host(
+                args.DATABRICKS_HOST).token(args.DATABRICKS_TOKEN).header(
+                    'x-databricks-session-id', session_id).getOrCreate()
+        else:
+            # IMPORTANT: make sure cluster has runtime newer than 14.1.0, the databricks-connect client version.
+            compute_id = args.cluster_id  # "1115-130834-ms4m0yv" - valid 14.1.0
+            w = WorkspaceClient()
+            res = w.clusters.get(
+                cluster_id=compute_id)  # '0704-124501-tsc2fxq' - invalid 12.2.x
+            runtime_version = res.spark_version.split('-scala')[0].replace(
+                'x-snapshot', '0').replace('x', '0')
+            if version.parse(runtime_version) < version.parse(
+                    MINIMUM_DBR_VERSION):
+                raise RuntimeError(
+                    f'You need at least {MINIMUM_DBR_VERSION} to use Databricks-connect to read delta table for FT API but got {res.spark_version}'
+                )
+            sparkSession = DatabricksSession.builder.remote(
+                host=args.DATABRICKS_HOST,
+                token=args.DATABRICKS_TOKEN,
+                cluster_id=compute_id).getOrCreate()
+
+    fetch(method, args.delta_table_name, args.json_output_path, args.batch_size,
+          args.processes, sparkSession, dbsql)
+
+    if dbsql is not None:
+        dbsql.close()
+
+    iterative_combine_jsons(
+        args.json_output_path,
+        os.path.join(args.json_output_path, 'combined.jsonl'))
+
+
+if __name__ == '__main__':
+    parser = ArgumentParser(
+        description=
+        'Download delta table from UC and convert to json to save local')
+    parser.add_argument('--delta_table_name',
+                        required=True,
+                        type=str,
+                        help='UC table <catalog>.<schema>.<table name>')
+    parser.add_argument('--json_output_path',
+                        required=True,
+                        type=str,
+                        help='Local path to save the converted json')
+    parser.add_argument('--DATABRICKS_HOST',
```

Comment:
heads up on these args, I believe we will need to get them from a databricks config file not env vars. cc @panchalhp-db knows more

Reply:
@panchalhp-db can you suggest what changes are appropriate here? Is there any code I can reuse to get the vars from config file?

Reply:
Hey sorry, I completely missed these github notifications. We decided that we will make the config file available at the path pointed to by `DATABRICKS_CONFIG_FILE` environment variable inside the user workload containers. You should be able to parse it to get the host and the token. @jbloxham would know the details of what the file would look like but I think it would likely be similar to what your local `~/.databrickscfg` file looks like.

Reply:
You should be able to use workspace client to get these creds eg:
```
from databricks.sdk import WorkspaceClient

w = WorkspaceClient()

host = w.config.host
token = w.config.token
```

Reply:
Yup, what Harsh said :)

Reply:
Thank you @panchalhp-db @jbloxham . Can you take another look at the latest change?